### PR TITLE
Update build badge

### PR DESCRIPTION
**Pull request by fsoubelet**
Fix for #51

### CODE REVIEW

**Comment by fsoubelet** [comment]
Will include fixes for #53 in here too

**Comment by fsoubelet** [comment]
I got the docstrings up-to-date in ef39b65, and in a format that goes well with `sphinx`'s parsing.

@lmalina can you check you're ok with how the doc looks now? Steps to get this version of the docs:

- Checkout this branch locally
- run `pip install ".[all]"
- build docs with `python -m sphinx -b html doc ./doc_build -d ./doc_build`
- open the newly created `doc_build` folder, then open `index.html` with your web browser

**Comment by fsoubelet** [comment]
I'm also including a quick fix in 2555247 for the type hint of `significant_digits`'s return, which was off by one element.

**Comment by fsoubelet** [comment]
Interestingly the build badge takes the name of the workflow it reports from, so for now on this branch this is `Cron Testing`. I could change the name to `CI Build` or something so that it looks a bit better on the README maybe.

@lmalina @JoschD ?

File: tfs/collection.py [hunk 1]
```diff
@@ -45,51 +45,58 @@ def __new__(mcs, cls_name, bases, dct: dict):
...
-    DataFrames.
+    classes inheriting from this abstract class will be able to define TFS files
+    as readable or writable, and read or write them just as attribute access or
+    assignments. All attributes will be read and written as `TfsDataFrames`.
 
     Example:
-        If `./example` is a directory that contains two TFS file `getbetax.out`
-        and `getbetax.out` with BETX and BETY columns respectively:
+        If **./example** is a directory that contains two TFS files `getbetax.tfs`
+        and `getbetax.tfs` with BETX and BETY columns respectively:
```
Comment (lmalina) [inline tfs/collection.py:57]: Put a real file names we use, for example, beta_phase_x.tfs and beta_phase_y.tfs

File: tfs/collection.py [hunk 2]
```diff
@@ -45,51 +45,58 @@ def __new__(mcs, cls_name, bases, dct: dict):
...
 
          class ExampleCollection(TfsCollection)
+            # All TFS attributes must be marked with the Tfs(...) class, and generated attribute
+            # names will be appended with _x / _y depending on files found in "./example"
 
-            # All TFS attributes must be marked with the Tfs(...) class:
-            beta = Tfs("getbeta{}.out")
-            # This is a traditional attribute.
-            other_value = 7
+            beta = Tfs("getbeta{}.tfs")  # A TFS attribute
```
Comment (lmalina) [inline tfs/collection.py:65]: same here

File: tfs/collection.py [hunk 3]
```diff
@@ -45,51 +45,58 @@ def __new__(mcs, cls_name, bases, dct: dict):
...
-            def get_filename(template, plane):
+            def get_filename(template: str, plane: str) -> str:
                return template.format(plane)
 
          example = ExampleCollection("./example")
-         # Get the BETX column from "getbetax.out":
-         beta_x_column = example.beta_x.BETX
-         # Get the BETY column from "getbetay.out":
+
+         # Get the BETX / BETY column from "getbetax.tfs":
```
Comment (lmalina) [inline tfs/collection.py:73]: same here

File: tfs/collection.py [hunk 4]
```diff
@@ -45,51 +45,58 @@ def __new__(mcs, cls_name, bases, dct: dict):
...
+         beta_x_column = example.beta_x.BETX  # / example.beta_x.BETY
+
+         # Get the BETY column from "getbetay.tfs":
```
Comment (lmalina) [inline tfs/collection.py:76]: and same here

File: tfs/handler.py [hunk 5]
```diff
@@ -337,15 +350,41 @@ def _parse_header(str_list: list) -> tuple:
...
+    `%s` identifiers only we might miss those looking like `%20s` specifying (here) a 20-character
+    wide column for strings.
+
+    Provided with a suspected identifier of this type, this function returns ``True`` if they are
+    in the `MAD-X` format, ``False`` otherwise.
+
+    The check is done by first checking that the identifier starts with `%` and finishes with `s`,
+    then by extracting the inner part of the string (for `%20s`, that would be `20`) and seeing if
+    it can be cast to an integer. If not, then the inner part is not indicating column width and
+    thus the suspicious identifier does not come from `MAD-X`.
```
Comment (lmalina) [inline tfs/handler.py:371]: Is this really needed? It duplicates the code

File: tfs/tools.py [hunk 6]
```diff
@@ -22,15 +22,15 @@
 
 def significant_digits(
     value: float, error: float, return_floats: bool = False
-) -> Tuple[Union[str, float]]:
+) -> Tuple[Union[str, float], Union[str, float]]:
```
Comment (lmalina) [inline tfs/tools.py:25]: ```Union[Tuple[str, str], Tuple[float,float]]``` , as we do not allow mixed types

**Review by lmalina — CHANGES_REQUESTED** [review]
nice, just few things to be changed ;-)

**Comment by fsoubelet** [comment]
All resolved in fd4a0a3. I also added a README file to the workflows directory so it's very clear which one does what.

@lmalina do we want this to be a new pre-minor version (`2.0.1`)? That would be an additional commit.

**Comment by lmalina** [comment]
> All resolved in [fd4a0a3](https://github.com/pylhc/tfs/commit/fd4a0a348832f92ab4f2edbbe0953b54f367c0e6). I also added a README file to the workflows directory so it's very clear which one does what.
> 
> @lmalina do we want this to be a new pre-minor version (`2.0.1`)? That would be an additional commit.

Yes, the absence of warning on MADX string column types is important to be released.

**Review by lmalina — APPROVED** [review]

**Pull request merged**